src/SpotifyBackend.cpp

```diff
@@ -0,0 +1,44 @@
+/** ----------------------------------------------------------------------------
+ * @file    SpotifyBackend.h
+ * @author  Stefan Jahn <stefan.jahn332@gmail.com>
+ * @brief   Class handles Music Playback with a Spotify Backend
+ * ---------------------------------------------------------------------------*/
+
+#include <vector>
+#include "SpotifyBackend.h"
+#include "Track.h"
+
+TResultOpt SpotifyBackend::initBackend() {
+    return std::nullopt;
+}
+
+TResult<std::vector<std::shared_ptr<Track>>> SpotifyBackend::queryTracks(std::string const &pattern, size_t const num) {
+    auto tmp = std::vector<std::shared_ptr<Track>>();
+    tmp.emplace_back(std::make_shared<Track>());
+    return tmp;
+}
+
+TResultOpt SpotifyBackend::setPlayback(std::shared_ptr<Track> const track) {
```

Comment:
do we need a whole track here? i mean the `MusicBackend` does not need more information than the track ID, correct? anyway, a const& should suffice

Reply:
yes, will change that